ldms/src/sampler/llnl/lustre_client.c

```diff
@@ -0,0 +1,641 @@
+
+        return set;
+}
+
+static void llite_stats_sample(const char *stats_path,
+                                   ldms_set_t general_metric_set)
+{
+        FILE *sf;
+        char buf[512];
+        int index;
+
+        sf = fopen(stats_path, "r");
+        if (sf == NULL) {
+                log_fn(LDMSD_LWARNING, SAMP" file %s not found\n",
+                       stats_path);
+                return;
+        }
+
+        /* The first line should always be "snapshot_time"
+           we will ignore it because it always contains the time that we read
+           from the file, not any information about when the stats last
+           changed */
+        if (fgets(buf, sizeof(buf), sf) == NULL) {
+                log_fn(LDMSD_LWARNING, SAMP" failed on read from %s\n",
+                       stats_path);
+                goto out1;
+        }
+        if (strncmp("snapshot_time", buf, sizeof("snapshot_time")-1) != 0) {
+                log_fn(LDMSD_LWARNING, SAMP" first line in %s is not \"snapshot_time\": %s\n",
+                       stats_path, buf);
+                goto out1;
+        }
+
+        ldms_transaction_begin(general_metric_set);
+        while (fgets(buf, sizeof(buf), sf)) {
+                uint64_t val1, val2;
+                int rc;
+                char str1[64+1];
+
+                rc = sscanf(buf, "%64s %lu samples [%*[^]]] %*u %*u %lu",
+                            str1, &val1, &val2);
+                if (rc == 2) {
+                        index = alt_ldms_metric_by_name(general_metric_set, str1);
+                        if (index == -1) {
+                                log_fn(LDMSD_LWARNING, SAMP" llite stats metric not found: %s\n",
+                                       str1);
+                        } else {
+                                ldms_metric_set_u64(general_metric_set, index, val1);
+                        }
+                        continue;
+                } else if (rc == 3) {
+                        int base_name_len = strlen(str1);
+                        sprintf(str1+base_name_len, ".sum"); /* append ".sum" */
+                        index = alt_ldms_metric_by_name(general_metric_set, str1);
+                        if (index == -1) {
+                                log_fn(LDMSD_LWARNING, SAMP" llite stats metric not found: %s\n",
+                                       str1);
+                        } else {
+                                ldms_metric_set_u64(general_metric_set, index, val2);
+                        }
+                        continue;
+                }
+        }
+
+	LJI_SAMPLE(general_metric_set, job_id_index);
+
+        ldms_transaction_end(general_metric_set);
+out1:
+        fclose(sf);
+
+        return;
+}
+
+static void llite_general_sample(const char *llite_name, const char *stats_path,
+                                 ldms_set_t general_metric_set)
+{
+        log_fn(LDMSD_LDEBUG, SAMP" llite_general_sample() %s\n",
```

Comment:
delete this call, unless you really want it and add a debug=<0/1> option to the sampler and put if (debug) around the logging call.

Reply:
The logging call almost certainly does exactly that check, I hope.  It must do something like "If current debug level < LDMSD_LDEBUG,  do nothing", right???

Reply:
The log call does check, but for a sampler that is coded correctly we do not want any per-sample debug logging unless it is 'opt-in'. It makes seeing things in the debug logs hard by default.

Reply:
If people want to use debug level logging, and they want to not see those messages from a particular plugin, they can simply not use that plugin while debugging, right?  Or if you really want to have different  log levels for different components, then it wouldn't be too difficult to add that functionality at logging library level rather than repeating it in each plugin.

Reply:
Please delete the DEBUG lines from the sample and llites_refresh. 
v3 is primarily in maintenance mode and per-plugin logging behaviors are not on the drawing board for it or for v4 (near release). v5 is just getting started internally. We don't use a separate logging library with the feature you suggest, but we  can discuss separately an extension to the logging API if you want to contribute one that trivially enables per-plugin control.